saleor/discount/models.py

```diff
@@ -55,22 +54,33 @@ def active_in_channel(self, date, channel_slug: str):
         )
 
     def expired(self, date):
-        return self.filter(
-            Q(used__gte=F("usage_limit")) | Q(end_date__lt=date), start_date__lt=date
+        return self.filter(end_date__lt=date, start_date__lt=date)
+
+
+class VoucherManager(models.Manager):
+    def get_queryset(self):
+        return VoucherQueryset(self.model, using=self._db).annotate(
+            used=Sum("codes__used")
```

Comment:
Do we need that annotate here? Did you check how the SQL looks like for this query?

Reply:
What about that?